### ISSUE TITLE
Does not work under Ubuntu 16.04 with Python 3.5.1

### ISSUE THREAD

**Issue by artizirk** [cjrh/aiorun#9]
Ubuntu 16.04 by default ships with Python 3.5.1 that has a `typing` library that does not provide `Coroutine` typing definition. This was fixed in Python 3.5.2 (python/typing#119) but afaik Ubuntu does not ship the fixed version.

Running the first example in the readme gives this output:
```
Traceback (most recent call last):
  File "./main.py", line 15, in <module>
    from aiorun import run
  File "/tmp/scratch/venv/lib/python3.5/site-packages/aiorun.py", line 14, in <module>
    from typing import Optional, Coroutine, Callable
ImportError: cannot import name 'Coroutine'
```

Im currently using this ugly hack and [typing library from pip](https://pypi.org/project/typing/) but a better solution should be figgured out.

```
# Ubuntu 16.04 ships with broken typing lib
# change sys path to use typing lib in virtualenv
if sys.version_info[:3] < (3, 5, 2):
    old_syspath = sys.path
    sys.path.insert(0, sys.path[-1])
    from aiorun import run
    sys.path = old_syspath
else:
    from aiorun import run
```

**Comment by cjrh** [comment]
@artizirk Thanks for the report! I had a quick look at the package list for Ubuntu 16.04. It looks like `python3` and `python3.5` packages are two different packages, and if you install `python3.5` you'll actually get `3.5.2` whereas if you install `python3` you are stuck with `3.5.1`.

- [Ubuntu Xenial (16.04) information for `python3` package](https://packages.ubuntu.com/xenial/python/python3)
- [Ubuntu Xenial (16.04) information for `python3.5` package](https://packages.ubuntu.com/xenial/python/python3.5)

Can your problem be resolved simply by using the `python3.5` package, as described in the link above?

**Comment by artizirk** [comment]
You are right, after enabling xenial-updates and xenial-security repositories I was able to install Python 3.5.2 but that did not fix the issue. 

I did some more digging and it looks like Coroutine typing definition was actually included starting from Python 3.5.3 
Relevant links:
* https://github.com/python/cpython/commit/62fe1bb983084c74fd8e7028412d0130a14568f3
* https://bugs.python.org/issue28556
* https://hg.python.org/cpython/rev/0201a87d773d

**Comment by cjrh** [comment]
Is `Coroutine` the only problem with 3.5.1/3.5.2? If so, I guess I could only import `Coroutine` under a version check, and then declare the typing signature using the string syntax for the type annotation. Would that be sufficient?

**Comment by artizirk** [comment]
Putting `Coroutine` import behind a version check should fix it.

**Comment by cjrh** [comment]
@artizirk The fix in #10 has now been push in `aiorun==2018.4.1`.  Sorry I took so long to finalize this. Thanks again for reporting!
